File: VintedProject/src/main/java/it/unical/ea/VintedProject/dto/BasicInsertionDto.java
package it.unical.ea.VintedProject.dto;


import jakarta.persistence.Basic;
import jakarta.persistence.FetchType;
import jakarta.persistence.Lob;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@ToString
public class BasicInsertionDto {

    @NotNull
    private Long id;

    @NotNull
    private String title;

    @NotNull
    private Float price;

    private String description;

    private String condition;

    @NotNull
    private LocalDate creationDate;

    private Boolean isPrivate;

    private LocalDate endDate;

    private String imageName;

    @NotNull
    private Long userId;
}

File: VintedAndroid/app/src/main/java/com/example/vintedandroid/dto/BasicInsertionDto.kt
package com.example.vintedandroid.dto

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDate

@Entity(tableName = "basicInsertionDto")
class BasicInsertionDto (
    @PrimaryKey val id : Long,
    @ColumnInfo val title : String,
    @ColumnInfo val price : Int,
    @ColumnInfo val description : String,
    @ColumnInfo val condition : String,
    @ColumnInfo val creationDate : LocalDate,
    @ColumnInfo val endDate: LocalDate,
    @ColumnInfo val imagePath : String,
    @ColumnInfo val userId : Long
        ){
}